demod_2400.c

```diff
@@ -25,6 +25,8 @@
 #include <gd.h>
 #endif
 
+#define M_SQRT2 1.41421356237
```

Comment:
This needs to come from an appropriate header or be #ifdef-ed. Same as M_PI.

Reply:
That code is guerded by `#if __BSD_VISIBLE || __XSI_VISIBLE`

You need to `#define __BSD_VISIBLE 1` directly or indirectly (perhaps on the command line)